### DISCUSSION TITLE
Clock / Calendar

### DISCUSSION THREAD

**Discussion by marcelo4768** [lawand-dot-io/taskbar#34] [Feature requests]
I would like to have a clock and access to the calendar on the far right

**Comment by lawand-dot-io** [comment]
Thanks for the request. I will consider adding that, but at the moment I feel like there's no need since the macOS's menubar is where they are located, so this would duplicate an existing feature.

**Reply by meirion** [reply]
I too would love the time displayed in bottom right. I feel like it's still muscle memory to glance there looking for the time - must be ingrained from those first days growing up on Windows 95!

I agree it's duplication of the mac clock top right however.

What is missing is a quick popup date grid calendar - even the Mac clock doesn't have this.

**Reply by lawand-dot-io** [reply]
@meirion thanks for the feedback, I see you commented on several feature requests 👍 there are already a couple of apps that add a calendar to the menubar, so I can't give this a high priority at the moment, but as always I will keep it open and the requests with the most upvotes are more likely to be implemented. cheers

**Reply by meirion** [reply]
Fair enough! 
It's led me to find 'Itsycal' which is great:
https://www.mowglii.com/itsycal/
<img width="281" alt="image" src="https://github.com/user-attachments/assets/e8f370fb-509f-42e6-8eea-d9df07b9ce53" />

Nice that Itsy cal shows the previous and next month days too - where as the Mac calendar desktop widget doesn't.

**Reply by lawand-dot-io** [reply]
@meirion thanks for the nice suggestion :)

**Reply by TheResplandor** [reply]
I hide MacOS' menu bar because i dont like it and its wasted screen space when using an external monitor (which is most of the time for me). Therefore i would love a simple clock on the side of the menu bar. A calender is much less needed but could also be nice.

In general I'm trying to use the menu bar as little as possible, and since taskbar is constantly there, quick things like clock (and battery/date/wifi/bluetooth) would ideally be there, like how things are in windows10/11 and KDE Plasma.

**Reply by aikpm** [reply]
> Thanks for the request. I will consider adding that, but at the moment I feel like there's no need since the macOS's menubar is where they are located, so this would duplicate an existing feature.

I have actually hidden the Mac top menu bar, so its more like Windows. I only have the Taskbar in view at all time. So I have to take the mouse and hover at the top of the screen for the menu bar to open and show me the battery percentage as well.

Appreciate if you can add the main windows taskbar icons on the right side of the taskbar: Battery, Time, Date, Wifi/Network

**Comment by lawand-dot-io** [comment]
Another alternative solution: make clicking on the clock in the menubar show a calendar widget:

<img width="375" alt="Screenshot 2025-01-25 at 00 36 08" src="https://github.com/user-attachments/assets/5eee9bdc-2633-4ff9-a640-edf4faadabfb" />

**Reply by janeznovakovslo-cmyk** [reply]
I really want to disable/hide the Menu Bar and only use the Mac Taskbar to navigate. So if at all possible I would greatly appreciate if you took the time and added date/time option in the bottom right like in Windows. Thank you! :)

**Comment by matt1** [comment]
+1 to missing clock being a major let-down versus uBar and others.

There is no macOs menu bar on external monitors where I do 99.9% of my work when I am forced to use a Mac, so not having it in taskbar is a disappointment and (in my opinion) a major missing feature.

Ideally we'd have a uBar-style/Windows-style right-aligned area for the clock and other system icons/widgets such as charge status, volume, network etc, but just starting with the clock would be a big quality of life improvement for this app.

**Reply by lawand-dot-io** [reply]
@matt1 thanks for this additional context, I will take it into consideration. Cheers

**Comment by carlosbrollotobar** [comment]
or a way to "pull" menu-bar items to the taskbar rigth side (windows like)
